diff --git a/library/ecp.c b/library/ecp.c
index 5c4464feb0e60254fad4183f83fe34f73046312e..f0f4feef04a54d1e80a8c8ac600d829b59d304e4 100644
--- a/library/ecp.c
+++ b/library/ecp.c
@@ -287,7 +287,7 @@ int esp_mbedtls_ecp_check_budget(const mbedtls_ecp_group *grp,
             if (rs_ctx->SUB == NULL)                                       \
             return MBEDTLS_ERR_ECP_ALLOC_FAILED;                     \
                                                                       \
-            ecp_restart_## SUB ##_init(rs_ctx->SUB);                      \
+            esp_ecp_restart_## SUB ##_init(rs_ctx->SUB);                      \
         }                                                                   \
 } while (0)
 
@@ -297,7 +297,7 @@ int esp_mbedtls_ecp_check_budget(const mbedtls_ecp_group *grp,
         if (rs_ctx != NULL && rs_ctx->SUB != NULL &&                        \
             ret != MBEDTLS_ERR_ECP_IN_PROGRESS)                            \
         {                                                                   \
-            ecp_restart_## SUB ##_free(rs_ctx->SUB);                      \
+            esp_ecp_restart_## SUB ##_free(rs_ctx->SUB);                      \
             esp_mbedtls_free(rs_ctx->SUB);                                    \
             rs_ctx->SUB = NULL;                                             \
         }                                                                   \
